presto-spi/src/main/java/com/facebook/presto/spi/hashing/ConsistentHashSelector.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.hashing;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.SortedMap;
+import java.util.TreeMap;
+
+import static com.google.common.hash.Hashing.md5;
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+public class ConsistentHashSelector<T>
+{
+    private final SortedMap<Long, T> circle = new TreeMap();
+    private static final HashFunction hashFunction = key -> md5().hashString(key, UTF_8).asLong();
```

Comment:
Looks like md5 is not recommended from the document, how about changing to `murmur3_32`